packages/mask/src/plugins/RedPacket/SNSAdaptor/hooks/useRedPacketContract.ts

```diff
@@ -6,20 +6,20 @@ import type { HappyRedPacketV1 } from '@masknet/web3-contracts/types/HappyRedPac
 import type { HappyRedPacketV2 } from '@masknet/web3-contracts/types/HappyRedPacketV2'
 import type { HappyRedPacketV3 } from '@masknet/web3-contracts/types/HappyRedPacketV3'
 import type { HappyRedPacketV4 } from '@masknet/web3-contracts/types/HappyRedPacketV4'
-import { useContract, useRedPacketConstants } from '@masknet/web3-shared-evm'
+import { useContract, useRedPacketConstants, ChainId } from '@masknet/web3-shared-evm'
 import type { AbiItem } from 'web3-utils'
 
-export function useRedPacketContract(version: number) {
+export function useRedPacketContract(version: number, chainId?: ChainId) {
     const {
         HAPPY_RED_PACKET_ADDRESS_V1: addressV1,
         HAPPY_RED_PACKET_ADDRESS_V2: addressV2,
         HAPPY_RED_PACKET_ADDRESS_V3: addressV3,
         HAPPY_RED_PACKET_ADDRESS_V4: addressV4,
-    } = useRedPacketConstants()
-    const v1 = useContract<HappyRedPacketV1>(addressV1, HappyRedPacketV1ABI as AbiItem[])
-    const v2 = useContract<HappyRedPacketV2>(addressV2, HappyRedPacketV2ABI as AbiItem[])
-    const v3 = useContract<HappyRedPacketV3>(addressV3, HappyRedPacketV3ABI as AbiItem[])
-    const v4 = useContract<HappyRedPacketV4>(addressV4, HappyRedPacketV4ABI as AbiItem[])
+    } = useRedPacketConstants(chainId)
+    const v1 = useContract<HappyRedPacketV1>(addressV1, HappyRedPacketV1ABI as AbiItem[], chainId)
```

Comment:
In the coming `refactor/migrate-evm`, the `chainId` goes the first parameter of all `useXXContract`. You could do the same thing in this PR too.

```suggestion
    const v1 = useContract<HappyRedPacketV1>(chainId, addressV1, HappyRedPacketV1ABI as AbiItem[])
```